src/core/middlewares/chart/builder.go

```diff
@@ -103,20 +103,23 @@ func (*chartVersionCreationBuilder) Build(req *http.Request) (interceptor.Interc
 		return nil, fmt.Errorf("project %s not found", namespace)
 	}
 
-	chart, err := parseChart(req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse chart from body, error: %v", err)
-	}
-	chartName, version := chart.Metadata.Name, chart.Metadata.Version
-
-	info := &util.ChartVersionInfo{
-		ProjectID: project.ProjectID,
-		Namespace: namespace,
-		ChartName: chartName,
-		Version:   version,
+	info, ok := util.ChartVersionInfoFromContext(req.Context())
```

Comment:
Not quite understand why we need to update the `chartVersionCreationBuilder ` instead of  `chartVersionDeletionBuilder ` as the issue #8701 is caused when deleting a chart